### PR TITLE
Multi TFMs build

### PR DESCRIPTION
**Pull request by jtkech**
@sebastienros 

As commented in #14732, if we apply #14732 and if we support multiple TFMs, when building by specifying a TFM it only copy the translation files if the specified TFM is the last one.

So here I tried the following changes, maybe worth to try them on your machine

- Here we don't set our `IsCrossTargetedBuild` in a props file but through a new Target.

    This new Target is not called when the Tests project is built, so our custom property is not set in this context.  In this context the translation files are copied per TFM but it was the case before.

- Then I simplified the condition of the Target copying the translations files.

If it doesn't fix the issue, I suggest to just copy the files per TFM by removing the condition, I don't think that it really matters as we skip unchanged files.

Note: In this PR I re-added net7.0 so that it is easier to test, will need to be removed if we apply it.

### CODE REVIEW

File: src/OrchardCore.Build/Dependencies.AspNetCore.props [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
     <!-- TFMs used to build the abstractions and modules, by convention the default TFM is at the first position -->
     <!-- In a cross-targeting build, some assets are only copied on the first TFM, by convention the default TFM -->
-    <CommonTargetFrameworks Condition="'$(CommonTargetFrameworks)' == ''">net8.0</CommonTargetFrameworks>
+    <CommonTargetFrameworks Condition="'$(CommonTargetFrameworks)' == ''">net8.0;net7.0;net6.0</CommonTargetFrameworks>
```
Comment (hishamco) [inline src/OrchardCore.Build/Dependencies.AspNetCore.props:12]: Did we bring the previous versions back?

Comment (jtkech) [reply src/OrchardCore.Build/Dependencies.AspNetCore.props:12]: No ;) This for testing race conditions

Comment (hishamco) [reply src/OrchardCore.Build/Dependencies.AspNetCore.props:12]: I see, but why the previous TFM are there?

Comment (jtkech) [reply src/OrchardCore.Build/Dependencies.AspNetCore.props:12]: Because race conditions happen when building multiple TFMs at the same time

**Comment by jtkech** [comment]
@sebastienros just for info

I know that you are investigating on this issue and possible other ones but I did some tests in this PR.

I had different issues, one of them when generating xml files on `GenerateDepsFile`, as I remember more often under Windows, but only when mutating `Properties` of inner projects even if done in a Target.

So for now I think we better have to always copy translation files without any condition knowing that we skip unchanged files.

Note: Would be good that they provide a property accessible by inner projects, to know that we are building an inner project.

**Comment by sebastienros** [comment]
@jtkech I updated my PR, then looked at this one, and I see you are arriving to the same conclusion as me. So many hours just for that ;)

**Comment by jtkech** [comment]
No problem, it was interesting, yes they should provide a property accessible by inner projects, to know that we are building an inner project.

**Comment by jtkech** [comment]
Close, see #14732

**Pull request closed without merging**